packages/grid/x-data-grid-premium/src/components/GridGroupingCriteriaCell.tsx

```diff
@@ -81,10 +92,10 @@ export const GridGroupingCriteriaCell = (props: GridGroupingCriteriaCellProps) =
           </IconButton>
         )}
       </div>
-      <span>
-        {formattedValue === undefined ? rowNode.groupingKey : formattedValue}
-        {!hideDescendantCount && filteredDescendantCount > 0 ? ` (${filteredDescendantCount})` : ''}
-      </span>
+      {cellContent}
+      {!hideDescendantCount && filteredDescendantCount > 0 ? (
+        <span style={{ whiteSpace: 'pre' }}> ({filteredDescendantCount})</span>
```

Comment:
I had to put it into separate `span` element to avoid wrapping: https://codesandbox.io/s/tender-leavitt-by8js4?file=/demo.tsx